app/src/test/java/com/oracle/javafx/scenebuilder/app/util/TextInputControlPasterTest.java

```diff
@@ -0,0 +1,106 @@
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the distribution.
+ *  - Neither the name of Oracle Corporation and Gluon nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.javafx.scenebuilder.app.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Map;
+import java.util.concurrent.FutureTask;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.oracle.javafx.scenebuilder.app.JfxInitializer;
+
+import javafx.application.Platform;
+import javafx.scene.control.TextField;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.DataFormat;
+
+public class TextInputControlPasterTest {
+
+    private TextInputControlPaster classUnderTest;
+    
+    @BeforeClass
+    public static void prepare() {
+       JfxInitializer.initialize();
+    }
+    
+    @Test
+    public void that_text_is_pasted_properly_using_workaround() {
+        TextField textField = new TextField();
+        classUnderTest = new TextInputControlPaster(textField);
+        classUnderTest.pasteWorkaroundOnMacOs("1234567890");
+        assertEquals("1234567890", textField.getText());
+        
+        textField.positionCaret(5);
+        classUnderTest.pasteWorkaroundOnMacOs("ABCD");
+        assertEquals("12345ABCD67890", textField.getText());
+        
+        textField.positionCaret(5);
+        textField.selectPositionCaret(9);
+        classUnderTest.pasteWorkaroundOnMacOs("");
+        assertEquals("1234567890", textField.getText());
+        
+        classUnderTest.pasteWorkaroundOnMacOs(null);
+        assertEquals("1234567890", textField.getText());
+    }
+    
+    @Test
+    public void that_workaround_is_applied_conditionally() throws Exception {
+        TextField textField = new TextField();
+        classUnderTest = new TextInputControlPaster(textField, ()->"Text2Paste");
```

Comment:
add spaces to lambda

Reply:
Same here, I'll see to get checkstyle on this.